hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/queries/AsNodeOperatorQueriesTest.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.suites.queries;
+
+import static com.hedera.services.bdd.junit.TestTags.CRYPTO;
+import static com.hedera.services.bdd.spec.HapiSpec.hapiTest;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getAccountBalance;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getAccountInfo;
+import static com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sleepFor;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+
+import com.hedera.services.bdd.junit.HapiTest;
+import com.hedera.services.bdd.junit.HapiTestLifecycle;
+import com.hedera.services.bdd.junit.OrderedInIsolation;
+import com.hedera.services.bdd.junit.support.TestLifecycle;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Tag;
+
+@Tag(CRYPTO)
+@OrderedInIsolation
+@HapiTestLifecycle
+@DisplayName("Node Operator Queries")
+/**
+ * A class with Node Operator Queries tests
+ */
+public class AsNodeOperatorQueriesTest extends NodeOperatorQueriesBase {
+
+    @Nested
+    @DisplayName("verify payer balance")
+    /**
+     * A class with Node Operator tests that verify the payer balance
+     */
+    class PerformNodeOperatorQueryAndVerifyPayerBalance {
+
+        @BeforeAll
+        static void beforeAll(@NonNull final TestLifecycle lifecycle) {
+            lifecycle.doAdhoc(createAllAccountsAndTokens());
+        }
+
+        @HapiTest
+        final Stream<DynamicTest> nodeOperatorQueryVerifyPayerBalanceForAccountBalance() {
+            // declare payer account balance variables
+            final AtomicLong currentPayerBalance = new AtomicLong();
+            final AtomicLong payerBalanceAfterFirstQuery = new AtomicLong();
+            final AtomicLong payerBalanceAfterSecondQuery = new AtomicLong();
+            return hapiTest(withOpContext((spec, log) -> {
+                final var initialBalance = getAccountBalance(PAYER).exposingBalanceTo(currentPayerBalance::set);
+                // perform paid query, pay for the query with payer account
```

Comment:
Using `withOpContext(...)` for all test operations makes these tests much more difficult to read. Why is the op context necessary? 

Can we do something like this instead?

```
return hapiTest(
    getAccountBalance(NODE_OPERATOR).payingWith(PAYER).signedBy(PAYER),
    sleepFor(3_000),
    getAccountBalance(PAYER).exposingBalanceTo(payerBalanceAfterFirstQuery::set)
    ...
    doAdhoc(() -> {
        Assertions.assertEquals(
                        currentPayerBalance.get(), 
                        payerBalanceAfterFirstQuery.get(), 
                        "Balances are not equal!");
        Assertions.assertEquals(
                        payerBalanceAfterFirstQuery.get(),
                        payerBalanceAfterSecondQuery.get(),
                        "Balances are not equal!");
    })
); 
```

Another thought, aren't the account balances known at the start of the test, since they are set up with `createAllAccountsAndTokens`? If the account balances are known, we could specify an expected amount instead of tracking balances with awkward atomic variables:

```
return hapiTest(
    getAccountBalance(NODE_OPERATOR).payingWith(PAYER).signedBy(PAYER)
          .hasTinyBars(<expected amount>),
    sleepFor(3_000),
    getAccountBalance(NODE_OPERATOR).payingWith(PAYER).asNodeOperator()
          .hasTinyBars(<expected amount>),
    ... 
);
```

Reply:
You are right, I changed the tests without using withOpContext(...) for all operations.

Regarding the atomic variables, I think it is better to not track concrete amounts of the Payer balances as it will be really difficult to track that and currently we can reuse the Payer account for many tests and still assert only whether it was or was not charged which is enough for these tests. It will be really difficult to track the concrete amounts in these tests and reuse the account and I suggest we leave the asserts this way.

Reply:
> currently we can reuse the Payer account for many tests

Reusing accounts across tests without resetting their state very likely means 1️⃣ we won't be able to make these new tests run in parallel, 2️⃣ a higher chance of test flakes, and 3️⃣ that these tests will be harder to debug, all because the tests are not independent of each other :( As is the case with some existing hapi tests in our repo, it's possible that a single test failure in these interrelated tests will require running the whole suite, or even multiple suites, to troubleshoot. We've spent a lot of time hunting these sorts of problems in CI. I would definitely prefer that entities (accounts etc.) be re-created in each test to keep the tests unrelated, even if it makes each test slower. 

That said, keeping the atomic vars is an option if other advantages outweigh the cons. I'm sure I haven't thought of everything 🤪 More thoughts?

Reply:
I agree. Tests must be independent. Only immutable objects can be reused.

Reply:
I absolutely agree that the best approach is to create and clean up the test data for each individual test but I have some concerns in this case.
First one is that by creating the test data for each separate test, in some tests we might exceed the limitation of 7 operations so that they become too many.
Apart from that, even if we only create Payer in each test, wouldn’t that flood the network with too many accounts, when we can reuse one or two accounts instead? In these tests, the only thing that is important is whether the Payer was charged or not - whether his balance at the beginning of the test was changed or not and in most cases it should not be changed as the node operator queries should be free of charge. Keep in mind that even if we create new Payer for each test, we still won’t need to strictly verify the amount of it balance, but only whether it was charged or not and we will still need to use some variables in order to verify that. I was thinking that if during the creation of the tests the Payer balance becomes insufficient, we could increase it so that it works for these specific tests but I might be wrong.
My second concern with the creation of the test data for each test is related to the data cleanup. If we are going to create separate test data for each test, then we will need to be able to properly clean it so that the state is refreshed for each test. If we do such cleanup anyway then it is ok and this one is not valid.
These were my thoughts while I created the tests, if you think that creating separate test data for each happy test case is better, then I will do them this way, just have in mind these details.

Reply:
Is it mandatory to run these tests in isolation and order? In the past, we had huge problems with test tasks that took forever to complete. One reason was that we could not run tests in parallel, which made us very sensitive to the topic.

I do not think you have to worry about the test data. These tests are executed against ad hoc networks, which are discarded after the tests have finished. For this reason, it is probably not even necessary to clean up the data afterward.

Why does it concern you that we may exceed seven operations? Is there a limit?

Reply:
Regarding the limit, I was referring to this requirement from the style guide checklist: https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/README.md#style-guide:~:text=The%20test%20includes%20no%20more%20than%20seven%20(7)%20SpecOperation%27s.
Regarding the clean up it is great, I was not aware, so this one was not valid.
It was necessary to run the tests in isolation exactly to avoid race conditions because of the reused test data. If I create the test data for each separate test that won't be necessary. 
In regard to that, I have added new commit with only test accounts for payer and node operator created for each test and removed the ordered in isolation tag, please let me know if this is what we are aiming.

Reply:
Thanks for making these tests independent, @Evdokia-Georgieva!! Our future selves will all thank us 😅